cmd/trace-agent/run.go

```diff
@@ -214,6 +217,45 @@ func Run(ctx context.Context) {
 		},
 	})
 
+	// prepare go runtime
+	agentrt.SetMaxProcs()
+	procs := runtime.GOMAXPROCS(0)
+	if mp, ok := os.LookupEnv("GOMAXPROCS"); ok {
+		log.Infof("GOMAXPROCS manually set to %v", mp)
+	} else if cfg.MaxCPU > 0 {
+		allowedCores := int(cfg.MaxCPU / 100)
+		if allowedCores < 1 {
+			allowedCores = 1
+		}
+		if allowedCores < procs {
+			log.Infof("apm_config.max_cpu is less than current GOMAXPROCS. Setting GOMAXPROCS to (%v) %d\n", allowedCores, (allowedCores))
+			runtime.GOMAXPROCS(int(allowedCores))
+		}
+	} else {
+		log.Infof("apm_config.max_cpu is disabled. leaving GOMAXPROCS at current value.")
+	}
+	log.Infof("Trace Agent final GOMAXPROCS: %v", runtime.GOMAXPROCS(0))
+
+	// prepare go runtime
+	maxmem, err := agentrt.SetGoMemLimit(coreconfig.IsContainerized())
+	if err != nil {
+		log.Infof("Couldn't set Go memory limit from cgroup: %s", err)
+	}
+
+	if cfg.MaxMemory > 0 && (maxmem == 0 || int64(cfg.MaxMemory) < maxmem) {
+		log.Infof("apm_config.max_memory: %vMiB", int64(cfg.MaxMemory)/(1024*1024))
+		maxmem = int64(cfg.MaxMemory)
+	}
+	if lim, ok := os.LookupEnv("GOMEMLIMIT"); ok {
+		log.Infof("GOMEMLIMIT manually set to: %v", lim)
+	} else if maxmem > 0 {
+		finalmem := int64(float64(maxmem) * 0.9) // leave some headroom
+		debug.SetMemoryLimit(int64(finalmem))
+		log.Infof("Maximum memory available: %vMiB. Setting GOMEMLIMIT to 90%% of max: %vMiB", maxmem/(1024*1024), finalmem/(1024*1024))
+	} else {
+		log.Infof("GOMEMLIMIT unconstrained.")
+	}
```

Comment:
Not sure why we'd do this. Calling `agentrt.SetGoMemLimit` at L240 should be enough as it sets gomemlimit to 90% of the cgroup limit. Am I missing something?

Reply:
Fixed.